### ISSUE TITLE
Better docs for model properties

### ISSUE THREAD

**Issue by neelnanda-io** [neelnanda-io/TransformerLens#97]
Make [this table](https://github.com/neelnanda-io/TransformerLens/blob/main/easy_transformer/model_properties_table.md) better and cover key info for model architecture - whether it uses parallel attn & MLPs, and what positional embedding it is.

Add text at the bottom documenting the models more qualitatively, can basically copy this glossary: https://docs.google.com/document/d/1WONBzNqfKIxERejrrPlQMyKqg7jSFW92x5UMXNrMdPo/edit#heading=h.chq47zvs9cii

I'd want to add a separate table with training info: include training dataset, number of tokens, whether they were trained with dropout, whether they have checkpoints, whether trained with weight decay.

**Comment by neelnanda-io** [comment]
Bonus for doing this in a way that can be automatically updated as more models are added, but this is less important

**Comment by geroldcsendes** [comment]
I guess you mean the [Model Properties](https://neelnanda-io.github.io/TransformerLens/model_properties_table.html). I find the column names confusing and I agree it would need a proper clean-up. In particular, I am confused about the `d_` and `n_` prefixes. What is the difference between `d_head` and `n_heads`? I am also confused why attention-only models have non-zero `d_mlp`.

I am new to TranformerLens but would like to contribute. If you could point out resources to me that contain metadata about the models available in the package, I could get started. 
> Bonus for doing this in a way that can be automatically updated as more models are added, but this is less important

What is the release process for models? I can think of a solution if you give me some guidance

**Comment by jbloomAus** [comment]
d_heads is the dimension of the attention heads in multi-headed attention and n_heads is the number of heads per layer. Non-zero d_mlp is a bug I guess, you can turn off mlp without setting it to zero, not the biggest deal. 

Models are added from hugging face typically and weights are accessed via api when we call `from_pretrained`. If you want to get a sense for what many of those parameters refer to, you could go here: https://neelnanda-io.github.io/TransformerLens/transformer_lens.html#module-transformer_lens.HookedTransformerConfig

The annotated transformer might be helpful too for a more general resource: https://nlp.seas.harvard.edu/2018/04/03/attention.html. 

Don't know how to source more meta-data automatically if it isn't on hugging face.

**Comment by neelnanda-io** [comment]
This function will [automatically get you the config](https://github.com/neelnanda-io/TransformerLens/blob/0ffcc8ad647d9e991f4c2596557a9d7475617773/transformer_lens/loading_from_pretrained.py#L659). I just used it to automatically generate that janky table, I agree it could be much better! We can create a script that automatically generates the model table, and just ask people to run it as part of a new model PR.

As Joseph says `d_` means the size of a dimension, `n_` means the number of something (eg number of heads or layers). It'd be good to add a key for this at the top or something? 

Other things I'd want to add:
* Whether the model uses parallel attention (currently just the Pythia models and NeoX)
* Whether the model was trained with dropout (currently just GPT-2)
* Which tokenizer was used

**Comment by mivanit** [comment]
Currently working on this in https://github.com/mivanit/transformerlens-model-table, will make a PR once that is more cleaned up (and I get around mixtral gated repo issues)

**Comment by jlarson4** [comment]
We now have both the [HookedTransformer interactive table](https://transformerlensorg.github.io/TransformerLens/generated/model_properties_table.html) and the [TransformerBridge Dynamic Table](https://transformerlensorg.github.io/TransformerLens/generated/transformer_bridge_models.html).

Both tables cover the primary requested information, and the TransformerBridge table has been extended to include Tokenizer details in the next release.

Due to the volume of models being supported now, getting individual training data for each model may be a bit out of scope. I will open a new issue specifically for that detail, in case a contributor wants to take a look at it.

**Comment by mivanit** [comment]
It looks like the [interactive table](https://transformerlensorg.github.io/TransformerLens/_downloads/ca4fd13e730f80d02ce0a9037546904c/model_properties_table_interactive.html) is broken due to missing jsonl data -- this might be my fault, as I originally contributed that. I may try to fix it and integrate some of the data from the transformerlens bridge table (text quality, verified status), as having the table be sortable/filterable by n_params, d_model, and other config values is still useful I think.

EDIT: I see the link issue may have already been fixed on dev in #1285, my bad

**Comment by jlarson4** [comment]
No worries @mivanit I appreciate the additional feedback and thoughts! It is fixed now on `dev`, it was just an issue with how the doc system was treating the HTML file. Because it is in the `_downloads` directory, creating it as a link was automatically adding the `download` attribute to the link. All I had to do to get it working was create a plain HTML text link. It'll be included in the next release. If you want to add filters and additional features to the interactive table, that would be awesome!
